lib/rules/at-rule-empty-line-before/index.mjs

```diff
@@ -138,20 +139,23 @@ const rule = (primary, secondaryOptions, context) => {
 				return;
 			}
 
-			// Fix
-			if (context.fix && context.newline) {
-				if (expectEmptyLineBefore) {
-					addEmptyLineBefore(atRule, context.newline);
-				} else {
-					removeEmptyLinesBefore(atRule, context.newline);
-				}
-
-				return;
-			}
-
 			const message = expectEmptyLineBefore ? messages.expected : messages.rejected;
+			const mutate = expectEmptyLineBefore ? addEmptyLineBefore : removeEmptyLinesBefore;
 
-			report({ message, node: atRule, result, ruleName });
+			// Fix
+			const fix = fixEmptyLineBefore.bind(null, {
+				node: atRule,
+				newline: context.newline,
+				mutate,
+			});
```

Comment:
[suggest] Using arrow functions instead of `.bind()` seems more readable for me:

```suggestion
			const fix = () => fixEmptyLineBefore({
				node: atRule,
				newline: context.newline,
				mutate,
			});
```

Reply:
I don't like how prettier format it but I don't mind the change.